File: backend/internal/model/booking.go
package model

import "time"

type ScheduleSetting struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Weekday   int       `gorm:"column:weekday;not null;check:weekday between 1 and 5" json:"weekday"`
	StartTime string    `gorm:"column:start_time;not null" json:"start_time"`
	EndTime   string    `gorm:"column:end_time;not null" json:"end_time"`
	IsActive  bool      `gorm:"column:is_active;default:true" json:"is_active"`
	CreatedAt time.Time `gorm:"column:created_at" json:"created_at"`
	UpdatedAt time.Time `gorm:"column:updated_at" json:"updated_at"`
}

func (ScheduleSetting) TableName() string {
	return "schedule_settings"
}

type Booking struct {
	ID              uint      `gorm:"primaryKey" json:"id"`
	CompanyName     string    `gorm:"column:company_name;not null;size:100" json:"company_name"`
	CompanyLocation string    `gorm:"column:company_location;not null;size:100" json:"company_location"`
	BookingDate     string    `gorm:"column:booking_date;not null;index" json:"booking_date"`
	BookingTime     string    `gorm:"column:booking_time;not null" json:"booking_time"`
	ContactName     string    `gorm:"column:contact_name;not null;size:50" json:"contact_name"`
	ContactEmail    string    `gorm:"column:contact_email;not null;size:255" json:"contact_email"`
	ContactPhone    string    `gorm:"column:contact_phone;not null;size:20" json:"contact_phone"`
	Notes           string    `gorm:"column:notes;type:text" json:"notes,omitempty"`
	Status          string    `gorm:"column:status;default:pending;check:status in ('pending','confirmed','rejected','completed','cancelled')" json:"status"`
	RejectReason    string    `gorm:"column:reject_reason;type:text" json:"reject_reason,omitempty"`
	CreatedAt       time.Time `gorm:"column:created_at;index:idx_bookings_created_at" json:"created_at"`
	UpdatedAt       time.Time `gorm:"column:updated_at" json:"updated_at"`
}

func (Booking) TableName() string {
	return "bookings"
}

type BookingResponse struct {
	ID              uint      `json:"id"`
	CompanyName     string    `json:"company_name"`
	CompanyLocation string    `json:"company_location"`
	BookingDate     string    `json:"booking_date"`
	BookingTime     string    `json:"booking_time"`
	ContactName     string    `json:"contact_name"`
	ContactEmail    string    `json:"contact_email"`
	ContactPhone    string    `json:"contact_phone"`
	Notes           string    `json:"notes,omitempty"`
	Status          string    `json:"status"`
	RejectReason    string    `json:"reject_reason,omitempty"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

File: frontend/src/api/booking.ts
import apiClient from './client'
import type { ApiResponse, PaginatedResponse, Booking, ScheduleSetting, SlotsResponse } from '@/types'

export interface CreateBookingRequest {
  company_name: string
  company_location: string
  booking_date: string
  booking_time: string
  contact_name: string
  contact_email: string
  contact_phone: string
  notes?: string
}

export interface UpdateBookingStatusRequest {
  status: 'pending' | 'confirmed' | 'rejected' | 'completed' | 'cancelled'
  reject_reason?: string
}

export interface UpdateScheduleSettingsRequest {
  slots: Omit<ScheduleSetting, 'id' | 'created_at' | 'updated_at'>[]
}

export const bookingApi = {
  async getSlots(date: string): Promise<ApiResponse<SlotsResponse>> {
    const response = await apiClient.get('/bookings/slots', { params: { date } })
    return response.data
  },

  async createBooking(data: CreateBookingRequest): Promise<ApiResponse<Booking>> {
    const response = await apiClient.post('/bookings', data)
    return response.data
  },

  async getBookings(params?: {
    page?: number
    page_size?: number
    status?: string
    search?: string
  }): Promise<ApiResponse<PaginatedResponse<Booking>>> {
    const response = await apiClient.get('/bookings', { params })
    return response.data
  },

  async getBooking(id: number): Promise<ApiResponse<Booking>> {
    const response = await apiClient.get(`/bookings/${id}`)
    return response.data
  },

  async updateBookingStatus(
    id: number,
    data: UpdateBookingStatusRequest
  ): Promise<ApiResponse<Booking>> {
    const response = await apiClient.put(`/bookings/${id}/status`, data)
    return response.data
  },

  async getScheduleSettings(): Promise<ApiResponse<ScheduleSetting[]>> {
    const response = await apiClient.get('/schedule')
    return response.data
  },

  async updateScheduleSettings(
    data: UpdateScheduleSettingsRequest
  ): Promise<ApiResponse<void>> {
    const response = await apiClient.put('/schedule', data)
    return response.data
  },
}
